lld/test/MachO/objc-category-merging-complete-test.s

```diff
@@ -0,0 +1,764 @@
+# REQUIRES: aarch64
+# RUN: rm -rf %t; split-file %s %t && cd %t
+
+## Create a dylib to link against(a64_file1.dylib) and merge categories in the main binary (file2_merge_a64.exe)
+# RUN: llvm-mc -filetype=obj -triple=arm64-ios-simulator -o a64_file1.o a64_file1.s
+# RUN: ld64.lld a64_file1.o -o a64_file1.dylib -dylib -arch arm64 -platform_version ios-simulator 14.0 15.0
```

Comment:
Common practice is to use`%lld` instead of `ld64.lld`. This way making changes to the base flags can affect all tests instead of needing to go into each test file to update. Flags can still be overridden if needed. 

Applies to every single line that directly uses `ld64.lld`

Reply:
My tests (assembly, matched patterns) are for ios-simulator. `%lld` is meant to be used for MacOS. I can't seem to be able to overwrite `-target_platform` if using `%lld` leading to test errors.

Reply:
Using `%no-arg-lld` with `-platform_version` might work, as with other LIT tests.

Reply:
I moved the tests to `macos`, and now using `%lld`, it turns out not to be difficult.